bridge/contracts/Federation.sol

```diff
@@ -96,22 +134,49 @@ contract Federation is Initializable, UpgradableOwnable {
         emit BridgeChanged(_bridge);
     }
 
+    function setBridgeNFT(address _bridgeNFT) external onlyOwner {
+      _setBridgeNFT(_bridgeNFT);
+    }
+
+    function _setBridgeNFT(address _bridgeNFT) internal {
+      require(_bridgeNFT != NULL_ADDRESS, "Federation: Empty NFT bridge");
+      bridgeNFT = INFTBridge(_bridgeNFT);
+      emit BridgeNFTChanged(_bridgeNFT);
+    }
+
+    function validateTransaction(bytes32 transactionId) internal view returns(bool) {
+      uint transactionCount = getTransactionCount(transactionId);
+      return transactionCount >= required && transactionCount >= members.length / 2 + 1;
+    }
+
+    /**
+      @notice Vote in a transaction, if it has enough votes it accepts the transfer
+      @dev Always return true
+      @param originalTokenAddress The address of the original token in the main chain
+      @param sender The address who solicited the cross token
+      @param receiver Who is going to receive the token in the side chain
+      @param number Could be the amount if tokenType == COIN or the tokenId if tokenType == NFT
+      @param blockHash The block hash of the transaction that occurs the cross event
```

Comment:
I'd maybe change to `The block hash in which the transaction with the cross event occurred`